parser/Parser.Test/test.cpp

```diff
@@ -0,0 +1,42 @@
+#include "pch.h"
+#include "../Parser/CAFF.h"
+
+TEST(ParserTest, TestNegSize) {
+    const char* input_file = "../caff_files/neg.caff";
+    bool failed = false;
+    FILE* f = fopen(input_file, "rb");
+    if (f == nullptr) {
+        std::cerr << "Failed to open file [" << input_file << "] for reading" << std::endl;
+        return;
+    }
+    CAFF caff;
+    Reader r(f);
+    try {
+        caff.read(r);
+    }
+    catch (ParsingException&) {
+        failed = true;
+    }
+    EXPECT_TRUE(failed);
+}
+
+TEST(ParserTest, TestPoc64) {
```

Comment:
```suggestion
// Poc64 contains very large width and height. Because of overflow (truncation) in multiplication, width*height*3 would still equal to content_size. Must fail.
TEST(ParserTest, TestPoc64) {
```